cdap-api/src/main/java/co/cask/cdap/api/artifact/PluginRegistry.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.artifact;
+
+import co.cask.cdap.api.templates.AdapterContext;
+import co.cask.cdap.api.templates.plugins.PluginProperties;
+
+import javax.annotation.Nullable;
+
+/**
+ * This interface provides methods to register plugin usage in a CDAP Program. The registered plugins will be
+ * available at the runtime of the Program.
+ */
+public interface PluginRegistry {
```

Comment:
For now I have renamed it to PluginConfigurer. We can move the methods to ProgramConfigurer in the future when all programs support Plugins.